pkg/idp/third_party.go

```diff
@@ -55,14 +55,14 @@ type Configuration struct {
 	// AuthURL is the authorize url, typically something like: https://example.org/oauth2/auth
 	// Should only be used when the OAuth2 / OpenID Connect server is not supporting OpenID Connect Discovery and when
 	// `provider` is set to `generic`.
-	// validate that this field is required only when Provider field == "generic"
-	AuthURL string `json:"auth_url" yaml:"auth_url" validate:"required_if=Provider generic"`
+	// validate that this field is required only when Provider field == "generic" and IssuerURL is not empty
+	AuthURL string `json:"auth_url" yaml:"auth_url" validate:"required_if=IssuerURL;required_if=Provider generic"`
```

Comment:
Shouldn't this be:
```suggestion
	AuthURL string `json:"auth_url" yaml:"auth_url" validate:"required_unless=IssuerURL,required_if=Provider generic"`
```

AFAICT to chain validators you need `,` (see [here](https://pkg.go.dev/github.com/go-playground/validator/v10#hdr-Multiple_Validators)). Also we should require the AuthURL is `IssuerURL` is empty (not the other way around).

I haven't tested this so please correct me.

Reply:
Got confused here also, you're right